deploy_tools/cli.py

```diff
@@ -338,6 +354,37 @@ def call(
     click.echo(result)
 
 
+@main.command(
+    short_help="Generates an encrypted keystore file. Creates a new account if no private key is provided."
+)
+@keystore_file_save_option
+@private_key_option
+def generate_keystore(keystore_path: str, private_key: str):
+    if path.exists(keystore_path):
+        raise click.BadOptionUsage(  # type: ignore
```

Comment:
Why do you have to ignore the type here?

Reply:
Please only ignore the type if there is a good reason you have to do that.

Reply:
Yes, we should update the `mypy` version.  The linting fails because the stubs of `mypy` for the untyped `click` are not working. This has been solved with the recent version. I tried it at least with `mypy==0.720` (current `0.701`) and it works.
The other usages of `BadOptionUsage` do not fail, because they live in not typed functions.

I wasn't involved into the dependency upgrade process so far. Can I just adjust the version number within the `constraints.txt` file?

Reply:
I would not do it manually, I usually do it like this, but there might be better ways:
 - use a clean venv and install everything with the constraints.txt `pip install -r requirements.txt -c constraints.txt`
 - Upgrade you package `pip install package --upgrade`
 - Check that everything works
 - Write the new constraints.txt `pip freeze > constraints.txt`
 - Check that you did not accidently include somthing that should not be in there, like the current package or other things that were installed by accident in you venv, and reove those manually.

Reply:
But this can be done in a new issue.